### PR TITLE
Update terms in code to use more inclusive language

### LINKED ISSUE #15522: Update terms in code to use more inclusive language
A quick `ack` of the repo shows `whitelist` and `blacklist` referenced in several places.  Let's update these references in our code and repo documentation (e.g. README) to use more inclusive terms.  Suggestions: `allowlist` or `blockedlist`. For references in third-party libs, we can see if there is an updated version of that lib and whether there is any blockers to switching to that version. 

See wp.me/paaHJt-15h-p2 for more.

### PR DESCRIPTION
**Pull request by ceyhun**
Fixes #15522

- Replaced `whitelist` references with `allowlist`
- Replaced `blacklist` references with `blocklist`
- Created a new xcdatamodel to rename some of `BlogSettings` entity properties that included those references
- Renamed `jetpack_whitelisted_ips_viewed` event to `jetpack_allowlisted_ips_viewed`
- Renamed `jetpack_whitelisted_ips_changed` event to `jetpack_allowlisted_ips_changed `

To test:

- Check Core Data migration: Install the app on top of an old version and check if it's launching without errors
- Update `Blocklist` and `Automatically Approve` under `Settings` -> `Discussions` and check if they are working correctly
- From a Jetpack connected site update `Allowlisted IP addresses` under `Jetpack Settings` and check if it's working correctly

PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift [hunk 1]
```diff
@@ -313,10 +313,10 @@ import Foundation
...
         case .jetpackDisconnectRequested:
             return "jetpack_disconnect_requested"
-        case .jetpackWhitelistedIpsViewed:
-            return "jetpack_whitelisted_ips_viewed"
-        case .jetpackWhitelistedIpsChanged:
-            return "jetpack_whitelisted_ips_changed"
+        case .jetpackAllowlistedIpsViewed:
+            return "jetpack_allowlisted_ips_viewed"
+        case .jetpackAllowlistedIpsChanged:
+            return "jetpack_allowlisted_ips_changed"
```
Comment (ceyhun) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:339]: Not sure if this is okay to just change the event names, what could be the better/proper way of doing so @leandroalonso?

Comment (leandroalonso) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:339]: Hi @ceyhun! Thanks for making those changes.

I think it's ok to change it for all user-facing changes. As for the analytic events this change needs to be made on both platforms (Android & iOS) to ensure consistency.

If that's the case, all good. :D

Comment (ceyhun) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:339]: It wasn't :D Just created a PR to update Android as well: https://github.com/wordpress-mobile/WordPress-Android/pull/14223

**Comment by ceyhun** [comment]
Hey there @emilylaguna, Leandro suggested assigning you for a more in-depth review. I would be happy if you could take a look 🙏

**Review by emilylaguna — APPROVED** [review]
I gave this a test and all seems to be working with the exception of the blocklist comment below:

- I noticed when I change the Site Settings > Discussion > Blocklist the values were not persisted to my sites (both Jetpack and WP.com). I don't think this is an issue with your PR since it looks like its happening on prod as well.  Can you give this a test, and make an issue if it's happening for you too?
- I'm not sure how to test the Jetpack allowlisted IPs but I do see the values persisted on my Jetpack site so I'll assume it's working correctly.

**Comment by ceyhun** [comment]
> * I noticed when I change the Site Settings > Discussion > Blocklist the values were not persisted to my sites (both Jetpack and WP.com). I don't think this is an issue with your PR since it looks like its happening on prod as well.  Can you give this a test, and make an issue if it's happening for you too?

Thanks @emilylaguna! I just resolved the conflicts and tested this as well. I can confirm that this is also happening to me :( both on prod and also on this PR.

**Review by emilylaguna — APPROVED** [review]
Tested using the steps, Also installed an older version of WPiOS and upgraded to this version with no issues. 

🚀

**Comment by ceyhun** [comment]
Thanks a lot @emilylaguna 🎉 Created an issue for the blocklist bug: https://github.com/wordpress-mobile/WordPress-iOS/issues/16101

**Pull request merged**